nomad/search_endpoint.go

```diff
@@ -107,8 +108,32 @@ func roundUUIDDownIfOdd(prefix string, context structs.Context) string {
 
 // PrefixSearch is used to list matches for a given prefix, and returns
 // matching jobs, evaluations, allocations, and/or nodes.
-func (s *Search) PrefixSearch(args *structs.SearchRequest,
-	reply *structs.SearchResponse) error {
+func (s *Search) PrefixSearch(args *structs.SearchRequest, reply *structs.SearchResponse) error {
+	aclObj, err := s.srv.ResolveToken(args.SecretID)
+	if err != nil {
+		return err
+	}
+
+	// Require either node:read or namespace:read-job
+	nodeRead := aclObj.AllowNodeRead()
```

Comment:
Won't this be a nil pointer dereference if you don't have ACLs enabled?

Reply:
Yup, and a test fails... yeesh... sorry for missing that!